pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -160,6 +185,43 @@ func (a *Operator) syncOperatorGroups(obj interface{}) error {
 	return nil
 }
 
+// pruneEmptyStrings removes any items from a list that are empty
+func pruneEmptyStrings(strings []string) []string {
+	prunedStrings := []string{}
+	for _, str := range strings {
+		if str != "" {
+			prunedStrings = append(prunedStrings, str)
+		}
+	}
+	return prunedStrings
+}
+
+// setDifference implements Set Difference: A - B
+// https://en.wikipedia.org/wiki/Complement_(set_theory)#In_programming_languages
+func setDifference(a, b []string) (diff []string) {
+	if len(a) == 0 {
+		return diff
+	}
+
+	if len(b) == 0 {
+		return a
+	}
+
+	// Find the Set Difference.
+	m := make(map[string]bool)
+
+	for _, item := range b {
+		m[item] = true
+	}
+
+	for _, item := range a {
+		if _, ok := m[item]; !ok {
+			diff = append(diff, item)
+		}
+	}
+	return diff
+}
+
 func (a *Operator) operatorGroupDeleted(obj interface{}) {
```

Comment:
nit: not part of your change, but `a` could have a better name?
Edit: I see `a` being used as a variable name for `operator` across multiple helper functions. If that's the norm followed, then you could keep it as is.